include/loop_tool/agent.h

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright (c) Facebook, Inc. and its affiliates.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+#pragma once
+
+#include "ir.h"
+#include "mutate.h"
+
+
+namespace loop_tool {
+
+class LoopTreeAgent{
+
+private:
+    LoopTree lt;
+    LoopTree::TreeRef cursor;
+
+public:
+    LoopTreeAgent(const LoopTree& lt, LoopTree::TreeRef cursor=0):lt(lt), cursor(cursor) {}
+
+    ~LoopTreeAgent() {}
+
+    std::vector<std::string> get_available_actions(){
```

Comment:
I think it might be useful to make all the actions into an `enum` (or `enum class`) so you can pass them around as arguments efficiently and be more strict about usage.  You can then convert them to strings for CompilerGym/human readability in a separate function